app/actors/hyrax/actors/collections_membership_actor.rb

```diff
@@ -75,19 +121,18 @@ def has_destroy_flag?(hash)
         # being created directly in that collection.  The permissions will not be applied to the work if the collection
         # type is configured not to allow that or if the work is being created in more than one collection.
         #
-        # @param [Hash] attributes_collection which was extracted from env[:member_of_collections_attributes] in create().
-        # e.g.
-        #   'attributes_collection' => {
-        #     '0' => { 'id' = '12312412'},
-        #   }
+        # @param env [Hyrax::Actors::Enviornment]
         #
         # Given an array of collection_attributes when it is size:
         # * 0 do not set `env.attributes[:collection_id]`
         # * 1 set `env.attributes[:collection_id]` to the one and only one collection
         # * 2+ do not set `env.attributes[:collection_id]`
         #
         # NOTE: Only called from create.  All collections are being added as parents of a work.  None are being removed.
-        def extract_collection_id(env, attributes_collection)
+        def extract_collection_id(env)
+          attributes_collection =
+            env.attributes.fetch(:member_of_collections_attributes) { nil }
```

Comment:
This needs to be able to extract the id from the new :member_of_collections_attributes OR from the old style :member_of_collection_ids.  

See diffs for method extract_collection_id (line 84) in the PR that switched from the old style :member_of_collection_ids to the new :member_of_collections_attributes...  https://github.com/samvera/hyrax/commit/85591cafa9e9935d18fd3b07cf86fb35cb9a73d4#diff-dcaa34b138208d26b5cc4a005542ad71